### PR TITLE
Avoid costly check on every candidate we check against.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @sharwell @Therzok

File: src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs [hunk 1]
```diff
@@ -187,7 +187,7 @@ private static bool ContainsUpperCaseLetter(string pattern)
                 }
             }
 
-            var isLowercase = !ContainsUpperCaseLetter(patternChunk.Text);
+            var isLowercase = patternChunk.IsLowercase;
```
Comment (sharwell) [inline src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs]: :bulb: Can inline?

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs]: thought about it.  but liked as is.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
@olegtk BTW, this shoudl be ported to editor impl.

**Comment by olegtk** [comment]
I will port it to the editor impl. Is it critical for 15.3 or can be later? We are in escrow 6PM today.

**Comment by CyrusNajmabadi** [comment]
it's not critical.  it's just a nice to have.

**Comment by CyrusNajmabadi** [comment]
@olegtk it's not critical.  it's just a nice to have.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Pull request merged**